### PR TITLE
Use template category sending_vehicle in pinpoint

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Use the template category's sending vehicle to decide what Pinpoint pool to use.

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/382

# Test instructions | Instructions pour tester la modification

Send sms with various template categories and verify that the appropriate pool is used.

# Release Instructions | Instructions pour le déploiement

Should be behind the feature flag `FF_TEMPLATE_CATEGORY`

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: app/clients/sms/aws_pinpoint.py [hunk 1]
```diff
@@ -32,6 +32,12 @@ def send_sms(self, to, content, reference, multi=True, sender=None, template_id=
         else:
             pool_id = self.current_app.config["AWS_PINPOINT_DEFAULT_POOL_ID"]
 
+        use_sc = str(template_id) in self.current_app.config["AWS_PINPOINT_SC_TEMPLATE_IDS"] or sending_vehicle == "short_code"
```
Comment (jimleroyer) [inline app/clients/sms/aws_pinpoint.py:35]: I like that we keep the env var for now. Should we remove it long term? 🤔

**Comment by jimleroyer** [comment]
Looks good so far. Tell me when it comes out of draft and I will approve. 👍

File: app/clients/sms/aws_pinpoint.py (hunk 1 above)
Comment (sastels) [reply app/clients/sms/aws_pinpoint.py:39]: 💯 yes - in this PR I'll bring in the feature flag and use that to do it the old way with `AWS_PINPOINT_SC_TEMPLATE_IDS` or the new way without it, basically.

**Review by jimleroyer — APPROVED** [review]
:shipit:

File: app/delivery/send_to_providers.py [hunk 2]
```diff
@@ -104,13 +105,20 @@ def send_sms_to_provider(notification):
 
         else:
             try:
+                if current_app.config["FF_TEMPLATE_CATEGORY"]:
+                    template_category_id = template_dict.get("template_category_id")
+                    template_category = dao_get_template_category_by_id(template_category_id) if template_category_id else None
```
Comment (andrewleith) [inline app/delivery/send_to_providers.py:110]: Do you actually need to do this lookup?  Would just checking `template_category_id` for `null` be the same check without the db call?

File: app/clients/sms/aws_pinpoint.py [hunk 3]
```diff
@@ -21,16 +21,21 @@ def init_app(self, current_app, statsd_client, *args, **kwargs):
...
         opted_out = False
         response = {}
 
-        use_shortcode_pool = (
-            str(template_id) in self.current_app.config["AWS_PINPOINT_SC_TEMPLATE_IDS"]
-            or str(service_id) == self.current_app.config["NOTIFY_SERVICE_ID"]
-        )
+        if self.current_app.config["FF_TEMPLATE_CATEGORY"]:
+            use_shortcode_pool = (
+                sending_vehicle == "short_code" or str(service_id) == self.current_app.config["NOTIFY_SERVICE_ID"]
```
Comment (andrewleith) [inline app/clients/sms/aws_pinpoint.py:32]: Would it be possible to get the `sending_vehicle` options into an `enum` or something similar?

**Review by andrewleith — COMMENTED** [review]
Looks good, just a couple questions!

File: app/delivery/send_to_providers.py (hunk 2 above)
Comment (sastels) [reply app/delivery/send_to_providers.py]: We need the category to get the sending vehicle from, but I've refactored and simplified this conditional:
```
template_category_id = template_dict.get("template_category_id")
if current_app.config["FF_TEMPLATE_CATEGORY"] and template_category_id is not None:
    sending_vehicle = dao_get_template_category_by_id(template_category_id).sms_sending_vehicle
else:
    sending_vehicle = None
```

File: app/clients/sms/aws_pinpoint.py (hunk 3 above)
Comment (sastels) [reply app/clients/sms/aws_pinpoint.py]: That's a good idea! I've added an enum
```
class SmsSendingVehicles(Enum):
    SHORT_CODE = "short_code"
    LONG_CODE = "long_code"
```
And used it everywhere we use the sending vehicle (including in the database model)

**Review by andrewleith — APPROVED** [review]
LGTM :shipit:

**Pull request merged**